src/main/java/hudson/plugins/mercurial/HgExe.java

```diff
@@ -346,6 +348,40 @@ public ProcStarter run(ArgumentListBuilder args) {
     }
 
     /**
+     * Gets the branch name of given revision number or of the current workspace.
+     * @param rev the revision to identify; defaults to current working copy
+     */
+    public @CheckForNull String branch(FilePath repository, @Nullable String rev) throws IOException, InterruptedException {
+        ArgumentListBuilder builder = new ArgumentListBuilder("id", "--branch");
+        if (rev != null)
+            builder.add("--rev", rev);
+        String branch = popen(repository, listener, false, builder).trim();
+        if (branch.isEmpty()) {
+            listener.error(Messages.HgExe_expected_to_get_a_branch_name_but_got_nothing());
+            return null;
+        }
+        return branch;
+    }
+
+    /**
+     * Gets the version of used Mercurial installation.
+     */
+    public @CheckForNull String version() throws IOException, InterruptedException {
```

Comment:
I had been using `Capability` for this purpose, rather than inspecting version numbers: basically trying to pass `-B` to `share`, and if it fails, remember that fact and do not try again (for this installation). Cannot now remember if there was a reason I did not directly check version numbers.